### PR TITLE
Basic migration guide

### PR DESCRIPTION
**Pull request by richfitz**
Was talking migration with Katy's guest and this felt useful

### CODE REVIEW

File: vignettes/migrating.Rmd [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+* "Restart" a particle filter
+* Multi-phase systems (not believed used except [sircovid](https://mrc-ide.github.io/sircovid))
+
+# New interface
+
+The new interface is completely different, following feedback about the discoverability and documentation of features within the package.  We've also generally tried to make the functions more self-contained, less magic, and easier to understand.  Some lesser-used functions have been removed entirely.  We apologise for the inconvenience that this will cause in migrating old code, but unfortunately this cannot be done automatically.
+
+Previously, dust generators were "[R6 classes](https://r6.r-lib.org/)", which were created using `$new()` and then interacted with using methods (e.g., `sys$run()`.  In contrast, `dust2` uses free functions to create systems and to interact with these systems once created.
+
+In this table, consider the dust system `sir`, which is available via `dust_example()` in both versions of the package.  In dust 1.0.0 could allocate a system with 10 particles, set the initial conditions, run it forward for 20 time units and return the state matrix by writing:
```
Comment (weshinsley) [inline vignettes/migrating.Rmd:40]:
```suggestion
In this table, consider the dust system `sir`, which is available via `dust_example()` in both versions of the package.  In dust 1.0.0 we could allocate a system with 10 particles, set the initial conditions, run it forward for 20 time units and return the state matrix by writing:
```

File: vignettes/migrating.Rmd [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+And in dust2 we could write
+
+```{r}
+sir <- dust_example("sir")
+sys <- dust_system_create(sir, pars = list(), time = 0, n_particles = 10)
+dust_system_set_state_initial(sys)
+dust_system_run_to_time(sys, 10)
+dust_system_state(sys)
+```
```
Comment (weshinsley) [inline vignettes/migrating.Rmd:56]: `dt = 1` is explicit in dust 1, and doesn't occur in the migrated version as it's the default - do we need to spell that out?

File: vignettes/migrating.Rmd [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+Supports OpenMP     | `$has_openmp()`       | (not yet implemented)
+
+1. Previously, the number of particles was much less well defined (running multiple groups was always implemented after support for running a single group).  `n_particles` now represents the number of particles *per group*.
+
+Some less used methods were removed entirely:
+
+* `$shape()`: previously returned the dimension structure among groups.  We do not currently support this concept and we do not believe anyone used it.
+* `$run_adjoint()`: this was to support proof-of-concept calculation of derivatives, which has moved into `dust_likelihood_last_gradient()`
+* `$set_index()`, `$index()`: the concept of an "index" was to support efficient running of the particle filter when that was implemented in R in the `mcstate` package.  We had two "indexes" to consider - one for saving trajectories and the other for the comparison to data.  With the particle filter now entirely written in C++ this is no longer needed.
+* `$set_stochastic_schedule()`: we had the concept a "stochastic schedule" to support mixed time models.  This schedule is now set via `dt` on construction.
```
Comment (weshinsley) [inline vignettes/migrating.Rmd:102]:
```suggestion
* `$set_stochastic_schedule()`: we had the concept of a "stochastic schedule" to support mixed time models.  This schedule is now set via `dt` on construction.
```

**Review by weshinsley — APPROVED** [review]
All good - couple of nitpicks below

File: vignettes/migrating.Rmd (hunk 2 above)
Comment (richfitz) [reply vignettes/migrating.Rmd:56]: thanks - I think this needs a whole section, I'll add one today

**Review by weshinsley — APPROVED** [review]
Looks good to me!

**Pull request merged**